### PR TITLE
Turn activity propagation on in more cases

### PR DESCRIPTION
**Pull request by davidfowl**
- Turn propagation on if Activity.Current is non-null

#36954

### CODE REVIEW

File: src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs [hunk 1]
```diff
@@ -26,8 +26,8 @@ public DiagnosticsHandler(HttpMessageHandler innerHandler) : base(innerHandler)
 
         internal static bool IsEnabled()
         {
-            //check if someone listens to HttpHandlerDiagnosticListener
-            return s_diagnosticListener.IsEnabled();
+            //check if someone listens to HttpHandlerDiagnosticListener of if there's a current activity
+            return Activity.Current != null || s_diagnosticListener.IsEnabled();
```
Comment (davidfowl) [inline src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs]: This check isn't as fast as I originally thought, but I haven't measured the overhead (plus it's about to make an HTTP request 😄 )

Comment (noahfalk) [reply src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs]: Yeah last I looked, it wasn't all that trivial to dig out an async local. Perf-impact aside it looked fine to me. I assume the various opt-out mechanisms you mentioned in the issue are still forthcoming.

Comment (davidfowl) [reply src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs]: Yep. Some of those require public API (which means an API review), so I wanted to make a draft of the core before going there. Also I need to write tests.

**Comment by stephentoub** [comment]
@davidfowl, are you still working on this?

**Comment by davidfowl** [comment]
Speaking to @lmolkova about who will work on it in preview7. So yea, it still needs to happen.

**Comment by davidfowl** [comment]
Closing in favor of #38280

**Comment by lmolkova** [comment]
somehow I missed all notifications for this one

**Pull request closed without merging**